File: tests/basic/function_rename_args.ml
open Optitrust
open Target
open Ast


let _ = Run.doc_script_cpp (fun _ ->

  !! Function_basic.rename_args ["x1"] [cFunDef "f"];

)
"
void f(int x){
 x = 10;
 int y;
 y = x;
 x = y;
}

int main(){}
"

let _ = Run.script_cpp (fun () ->

  !! Function_basic.rename_args [] [cFunDef "test_no_args"];

  !! Function_basic.rename_args [""] [cFunDef "test_one_arg"];
  !! Function_basic.rename_args ["x1"] [cFunDef "test_one_arg"];

  !! Function_basic.rename_args ["x1"; ""] [cFunDef "test_two_args"];
  !! Function_basic.rename_args [""; "y1"] [cFunDef "test_two_args"];
  !! Function_basic.rename_args ["x2"; "y2"] [cFunDef "test_two_args"];

  !! Function_basic.rename_args ["x1"] [cFunDef "test_method_args"];
)

File: tests/basic/function_rename_args.cpp


void test_no_args(){
  int z = 10;
  z = 20;
}


void test_one_arg(int x){
  int z = x;
  z = 10;
}


void test_two_args(int x, int y){
  int z = x;
  for(int x = 0; x < 4; x++){
    z += x + y;
    int y = 1;
    int r = y + x;
  }
  int a = x + y;
}
